library/src/main/kotlin/com/ryanharter/ktor/moshi/MoshiConverter.kt

```diff
@@ -22,7 +24,9 @@ class MoshiConverter(private val moshi: Moshi = Moshi.Builder().build()) : Conte
     val channel = request.value as? ByteReadChannel ?: return null
     val source = channel.toInputStream().source().buffer()
     val type = request.type
-    return moshi.adapter(type.javaObjectType).fromJson(source)
+    return withContext(Dispatchers.IO) {
```

Comment:
I don't we should change the coroutine context. It is already a suspending function, so it will pick up the given context when executed.

Reply:
I bumped into this and did some investigation. Because `fromJson()` throws `IOException`, Intelij assumes it is doing a blocking call and warns that it is not safe to do in a coroutine. Using `Dispatchers.IO` silences this warning because it  is ["designed specifically to handle blocking calls"](https://discuss.kotlinlang.org/t/warning-inappropriate-blocking-method-call-with-coroutines-how-to-fix/16903/2).

I wasn't able to find any information on what ktor uses for `ContentConverter`s. It is possible that it is already on using `Dispatchers.IO`, which mean calling it directly would be fine. But the IDE inspection doesn't have that context to know if it is a false positive.

I looked at the implementation of the of the `ContentConverter`s for `gson` and `serialization`. Neither use `IOException`, so it is hard to tell if this is needed.